src/source.cpp

```diff
@@ -170,10 +172,21 @@ SourceSite Source::sample_with_constraints(uint64_t* seed) const
       accepted = satisfies_spatial_constraints(site.r) &&
                  satisfies_energy_constraints(site.E) &&
                  satisfies_time_constraints(site.time);
-      if (!accepted && rejection_strategy_ == RejectionStrategy::KILL) {
-        // Accept particle but set weight to 0 so that it is killed immediately
-        accepted = true;
-        site.wgt = 0.0;
+      if (!accepted) {
+        ++n_reject;
```

Comment:
I believe this is not thread safe use of a static variable, although the probability of two threads simultaneously hitting it is low. I suppose that having these be 32 bit types means you'll never get unreasonable values. Maybe the counters should be confined to a single thread? I know it doesn't influence correctness but worth a little thought.

Reply:
I also had this realization looking through the code and thought about how much we should care about it (the same thing applies in the existing logic in `IndependentSource`). The important thing is to trigger the condition when too many sites are getting rejected; even if there is a race condition on incrementing the number of accepts/rejects, that will still happen, just in a semi-fickle way. I was going to defer fixing it for now but if you think we should take care of it as part of this PR, I'm happy to do so. Let me know how I should proceed.

Reply:
Sounds good! I'm just going to raise a separate issue to handle it.